### PR TITLE
Upgrading flow to the last version and fixing flow errors

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Upgrading flow to the last version and fixing flow errors

#### Ticket Link
[MM-13322](https://mattermost.atlassian.net/browse/MM-13322)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed

### CODE REVIEW

File: src/utils/emoji_utils.js [hunk 1]
```diff
@@ -29,22 +29,24 @@ export function parseNeededCustomEmojisFromText(text: string, systemEmojis: Map<
 
     let match;
     while ((match = pattern.exec(text)) !== null) {
-        if (systemEmojis.has(match[1])) {
+        if (match && systemEmojis.has(match[1])) {
```
Comment (cometkim) [inline src/utils/emoji_utils.js]: it could be one `if (!match) continue;`

Comment (jespino) [reply src/utils/emoji_utils.js]: Fixed! Thanks :)

File: src/selectors/entities/common.js [hunk 2]
```diff
@@ -18,11 +18,11 @@ export function getMyChannelMemberships(state: GlobalState): {[string]: ChannelM
     return state.entities.channels.myMembers;
 }
 
-export const getMyCurrentChannelMembership = createSelector(
+export const getMyCurrentChannelMembership: (GlobalState) => ?ChannelMembership = createSelector(
     getCurrentChannelId,
     getMyChannelMemberships,
     (currentChannelId, channelMemberships) => {
-        return channelMemberships[currentChannelId] || {};
+        return channelMemberships[currentChannelId] || null;
```
[the remark this answers is not in the document]
Comment (jespino) [reply src/selectors/entities/common.js]: Yes, probably we need to review the webapp and mobile code before we merge it.

[the remark this answers is not in the document]
Comment (jespino) [reply src/selectors/entities/common.js]: ok, this leaks until the plugins, so, is outside our full control. I can change it and maybe, break anything in the header plugins. and I would need to fix some small things in mobile and webapp, but they are things that now are working in a weird way. Basically if you don't have the membership, it return "undefined" in some places, with is something that works, but is not ideal.

[the remark this answers is not in the document]
Comment (jespino) [reply src/selectors/entities/common.js]: Done, Fixed on webapp and mobile in the PRs: mattermost/mattermost-mobile#2488 and mattermost/mattermost-webapp#2238

**Pull request merged**